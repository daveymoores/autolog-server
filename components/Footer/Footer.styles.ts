import css from "styled-jsx/css";

export default css`
  footer {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: stretch;

    max-width: var(--max-content-width);
    width: 100%;
    height: auto;
    margin: 0 auto;
  }

  @media (prefers-color-scheme: dark) {
    footer {
      border: var(--lineWidth) solid var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    footer {
      border: var(--lineWidth) solid var(--darkGrey);
    }
  }

  .bottom-bar {
    height: var(--cell-height);
    display: flex;
    justify-content: space-between;
    align-items: stretch;
  }

  @media (prefers-color-scheme: dark) {
    .bottom-bar {
      border-top: var(--lineWidth) solid var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    .bottom-bar {
      border-top: var(--lineWidth) solid var(--darkGrey);
    }
  }

  ul {
    display: flex;
    justify-content: center;
    align-items: stretch;

    list-style: none;

    padding: 0;
    margin: 0 0 0 auto;
  }

  li {
    border-top: none;
    border-bottom: none;
    border-right: none;
  }

  li:nth-child(2) {
    padding: 0.7em 0.7em 0.7em 0;
    border-left: none;
  }
`;
