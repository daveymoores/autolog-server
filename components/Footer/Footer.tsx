import Image from "next/image";
import Link from "next/link";
import React from "react";

const Footer: React.FC = () => {
  return (
    <footer className="bg-black mt-24">
      <nav className="container font-semibold flex justify-center flex-col align-middle items-center gap-2 py-16">
        <Link href="/documentation">Github</Link>
        <Link href="/documentation">Linkedin</Link>
        <div className="mt-24">
          <Link href="/" passHref>
            <Image
              src="/logo.svg"
              width={114}
              height={15}
              priority={true}
              layout="fixed"
            />
          </Link>
        </div>
      </nav>
    </footer>
  );
};

export default Footer;
