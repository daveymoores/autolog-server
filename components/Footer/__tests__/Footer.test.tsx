import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";
import React from "react";

import Footer from "../Footer";

// Mock only the router, which is essential for the conditional rendering
jest.mock("next/router", () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

// We don't need to mock Next/Image in detail since we're not testing its behavior
// Just make it render something we can detect
jest.mock("next/image", () => ({
  __esModule: true,
  default: () => <div data-testid="logo-image" />,
}));

// Link doesn't need complex mocking either for basic tests
jest.mock("next/link", () => {
  return ({ children }: { children: React.ReactNode }) => children;
});

import { NextRouter, useRouter } from "next/router";
const mockUseRouter = useRouter as jest.MockedFunction<typeof useRouter>;

describe("Footer Component", () => {
  it("renders links when not in print mode", () => {
    mockUseRouter.mockReturnValue({ query: {} } as NextRouter);

    render(<Footer />);

    expect(screen.getByText("Github")).toBeInTheDocument();
    expect(screen.getByText("Linkedin")).toBeInTheDocument();
    expect(screen.getByTestId("logo-image")).toBeInTheDocument();
  });

  it("hides links in print mode", () => {
    mockUseRouter.mockReturnValue({
      query: { print: "true" },
    } as unknown as NextRouter);

    render(<Footer />);

    expect(screen.queryByText("Github")).not.toBeInTheDocument();
    expect(screen.queryByText("Linkedin")).not.toBeInTheDocument();
    expect(screen.getByTestId("logo-image")).toBeInTheDocument();
  });
});
