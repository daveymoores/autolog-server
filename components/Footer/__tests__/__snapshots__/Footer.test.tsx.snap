// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer unchanged 1`] = `
<footer
  className="jsx-1890663278"
>
  <div
    className="jsx-591473389 newsletter-subscribe"
  >
    <h3
      className="jsx-591473389"
    >
      Subscribe to be the first to know about updates
    </h3>
    <form
      className="jsx-591473389"
    >
      <input
        className="jsx-353473093 t-gen--cell"
        type="text"
      />
      <button
        className="jsx-836649993"
        type="submit"
      >
        subscribe
      </button>
    </form>
  </div>
  <div
    className="jsx-1890663278 bottom-bar"
  >
    <ul
      className="jsx-1890663278"
    >
      <li
        className="jsx-1890663278 t-gen--cell"
      >
        <a
          className="jsx-1890663278"
          href=""
        >
          View on Github
        </a>
      </li>
      <li
        className="jsx-1890663278 t-gen--cell"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMjUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="GitHub icon"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="GitHub icon"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2FGitHub-Mark-Light-64px.svg&w=64&q=75"
              srcSet="/_next/image?url=%2FGitHub-Mark-Light-64px.svg&w=32&q=75 1x, /_next/image?url=%2FGitHub-Mark-Light-64px.svg&w=64&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </li>
    </ul>
  </div>
</footer>
`;
